### PR TITLE
added front end pagination

### PR DESCRIPTION
**Pull request by priyatharsan**
closes https://github.com/plotly/dash-table/issues/266

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,6 +6,7 @@ coverage/
 dash_core_components/metadata.json
 dash_core_components/bundle.js*
 node_modules/
+cache-directory/
```
Comment (cldougl) [inline .gitignore]: @priyatharsan why was this modified in this pr?

Comment (priyatharsan) [reply .gitignore]: It came into my attention that `cache-directory` I suppose a temporary folder popped up and realized that it may not be required in the repo. Thought I could fix it in this PR itself, but I shall revert that and create a new PR? @cldougl

Comment (cldougl) [reply .gitignore]: yes please :)
